packages/front-end/components/Settings/EditDataSource/ExperimentAssignmentQueries/AddEditExperimentAssignmentQueryModal.tsx

```diff
@@ -111,7 +110,7 @@ export const AddEditExperimentAssignmentQueryModal: FC<EditExperimentAssignmentQ
     return null;
   }
 
-  const validateResponse = (result: TestQueryResults) => {
+  const validateResponse = (result: TestQueryRow) => {
```

Comment:
Don't include the "useNameColumns" or "dimensions" as required columns when passing into the EditSQL modal

Reply:
This approach is problematic.  We are doing a check after save and warning whether they have missing columns on the main data source page.  If I remove the check there as well, then users who have manually added a dimension in the form and hit save might not have a sql statement that actually uses it.  

Also if they click "use name column" but we don't show them then as required columns, then our users wouldn't know the proper alias to use.  

I think the current experience as well isn't the best UX.  Like the click the "use name column", edit sql, don't modify their query, hit save, and then we unselect use name column for them.  Or they manually add a dimensions, edit sql, don't add the column, and then their dimension they manually added is gone.  

So I plan on leaving the current behavior by renaming validateResponse to validateResponseOverride.  Then in EditSql modal, it either calls the override function if it exists, otherwise it calls its own one to check the required columns.

Longer term I would suggest we stop automatically removing "use name columns" nor automatically removing dimensions.  Or get rid of dimensions on the form all together because as we have it now the sql response is really the source of truth for dimensions. Perhaps we get rid of the toggle for "use name columns" as well.  We would just need to always run the query when saving and update the datasource object appropriately, and also make it clear to the user in some message somewhere that the extra columns become experiment dimensions (and what experiment dimensions are) and that experiment_name and variation_name are optional.